Enable crsf and ghst telemetry on supported boards

benjinne opened the pull request:
**Describe problem solved by this pull request**
See issue https://github.com/PX4/PX4-Autopilot/issues/17232

**Describe your solution**
I added BOARD_SUPPORTS_RC_SERIAL_PORT_OUTPUT define to the supported board configs and changed the check for CONFIG_ARCH_BOARD_OMNIBUS_F4SD to BOARD_SUPPORTS_RC_SERIAL_PORT_OUTPUT in RCInput.cpp.

**Test data / coverage**
Tested on modalaiFC-V1 board with a crossfire receiver sending telemetry to my radiomaster TX16s transmitter

dagar commented:
Looks good, although longer term it would be nice if we could do this automatically based on what's already defined for the board.

dagar commented:
Would you mind adding this around ghost telemetry as well?

https://github.com/PX4/PX4-Autopilot/blob/4349470a14171a29828da951eee4ac119d39eb5a/src/drivers/rc_input/RCInput.cpp#L675-L677

dagar commented:
Thanks @benjinne

benjinne commented:
> Would you mind adding this around ghost telemetry as well?
> 
> https://github.com/PX4/PX4-Autopilot/blob/4349470a14171a29828da951eee4ac119d39eb5a/src/drivers/rc_input/RCInput.cpp#L675-L677

Done, and added the macro to fmu-v5 since a comment claims it also works. I also updated some comments
